src/net.cpp

```diff
@@ -2501,7 +2501,7 @@ void CConnman::Interrupt()
     }
 }
 
-void CConnman::Stop()
+void CConnman::Stop() NO_THREAD_SAFETY_ANALYSIS
```

Comment:
Why?

Reply:
Good question! I added a clarifying comment.

The problem here is that `cs_vNodes` needs to be called before reading the variable `vNodes`.

Adding such a lock in `CConnman::Stop` would be trivial, but due to the implicit locking order requirements that follow from other parts of the code getting a matching locking order here results in surprisingly invasive changes (at least when I did it!).